### PR TITLE
feat: check stability of output

### LINKED ISSUE #170: catch race conditions
In case of race condition inside a kernel, output will change from generation to generation.
Attention kernel has a high risk of race condition.
We may want to add a specific check for that purpose.

### PR DESCRIPTION
**Pull request by pommedeterresautee**
fix #170 

Sometimes we have bugs on masks, or race condition issues.
They may be catched by checking if the output of the kernel change in time or not.
This is inspired from https://github.com/HazyResearch/flash-attention/blob/main/tests/test_flash_attn.py#L978
The loop is smaller to not make tests too slow

Moreover, old implementations are removed as our kernel is now quite different, old implementations are of little value and make tests slower.

### CODE REVIEW

**Comment by pommedeterresautee** [comment]
```
❯ pytest test/test_attention.py
===================================================================================================== test session starts =====================================================================================================
platform linux -- Python 3.9.15, pytest-7.2.0, pluggy-1.0.0
rootdir: /mnt/workspace/kernl
plugins: anyio-3.6.1
collected 699 items                                                                                                                                                                                                           

test/test_attention.py ................................................................................................................................................................................................ [ 27%]
....................................................................................................................................................................................................................... [ 58%]
....................................................................................................................................................................................................................... [ 88%]
.............................................................................                                                                                                                                           [100%]
...
=============================================================================================== 699 passed in 723.90s (0:12:03) ===============================================================================================

```

File: test/test_attention.py [hunk 1]
```diff
@@ -106,6 +90,9 @@ def test_benchmark_masked(
     func = implementations[implementation]
     value = benchmark(func, **cast_args)
 
+    for _ in range(10):
```
Comment (gaetansnl) [inline test/test_attention.py:93]: put this after for early fail ?

**Review by gaetansnl — APPROVED** [review]
single comment

File: test/test_attention.py (hunk 1 above)
Comment (gaetansnl) [reply test/test_attention.py:93]: and do we need that much run ?

Comment (gaetansnl) [reply test/test_attention.py:93]: and can we have a different message for output stability ?

Comment (pommedeterresautee) [reply test/test_attention.py:104]: * ok for early fail
* on Hazyresearch they put 10k, I think it matters to repeat stuff
* message for output stability would require to change the assert function, and right now we have enough info (we have the breaking line in pytest output), seems to be ok to me

**Pull request merged**